src/main/java/com/github/_1c_syntax/bsl/languageserver/diagnostics/reporter/JUnitTestSuites.java

```diff
@@ -60,7 +60,7 @@
   @JacksonXmlElementWrapper(useWrapping = false)
   private final List<JUnitTestSuite> testsuite;
 
-  private static Path sourceDir;
+  private static Path sourceDir = Paths.get(".");
```

Comment:
Хм. Мне кажется, что проблема не в дефолтном значении, а в том, что в 67 строке ты в объекте переопределяешь статик-поле. Оно точно так задумывалось? Может это поле объекта, а не класса?

Reply:
я не совсем понимаю, как сделать: это поле содержит месторасположение исходников, которые сканировались. только статиком я сделал его доступным во вложенных классах. Возможно я не прав, и это делается иначе

Reply:
Это потому что у тебя JUnitTestSuite является статик-классом. Если убрать static из объявления, тогда он сможет обращаться к переменным экземпляра внешнего класса.

Reply:
inner class may be static

Reply:
may. а может и не may :) если он static, то ему недоступны поля экземпляра. если не-static - доступны.

Reply:
ну как бы говорят что

> The basic reason for Jackson's refusal to try use non-static inner classes for deserialization (serialization actually works fine) is because there is no general way to instantiate such classes -- there is no zero-argument constructor, nor @JsonCreator annotated other constructors or factory methods (or single-String-argument constructors, but I digress). Jackson can not instantiate them.

ну а идея говорит 
```
Cannot construct instance of `com.github._1c_syntax.bsl.languageserver.diagnostics.reporter.JUnitTestSuites$JUnitTestSuite` (although at least one Creator exists): can only instantiate non-static inner class by using default, no-argument constructor
 at [Source: (File); line: 2, column: 3] (through reference chain: com.github._1c_syntax.bsl.languageserver.diagnostics.reporter.JUnitTestSuites["testsuite"]->java.util.ArrayList[0])
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `com.github._1c_syntax.bsl.languageserver.diagnostics.reporter.JUnitTestSuites$JUnitTestSuite` (although at least one Creator exists): can only instantiate non-static inner class by using default, no-argument constructor
 at [Source: (File); line: 2, column: 3] (through reference chain: com.github._1c_syntax.bsl.languageserver.diagnostics.reporter.JUnitTestSuites["testsuite"]->java.util.ArrayList[0])
```

Подскажи, как тут можно решить, пока я не догадался

Reply:
О, это другой разговор. Покумекаю